### PR TITLE
Adding Test script to validate the python result for different experiment result

### PR DESCRIPTION
**Pull request by shruacha1234**
Signed-off-by: Shruthi <Shruthi.Acharya@ibm.com>

### CODE REVIEW

**Comment by chandrams** [comment]
@shruacha1234 - Can you please resolve the conflicts

File: tests/scripts/rm_hpo_api_tests.sh [hunk 1]
```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
...
 # Tests for RM-HPO POST experiment
 function rm_hpo_post_experiment() {
-	invalid_post ${FUNCNAME}
+	run_post_tests ${FUNCNAME}
 	other_post_experiment_tests ${FUNCNAME}
 }
+
+# Tests for RM-HPO POST experiment
+function rm_hpo_post_exp_result() {
+	valid_exp="valid-experiment"
```
Comment (chandrams) [inline tests/scripts/rm_hpo_api_tests.sh:491]: Initialize these variables in the functions where they are used. Also add comments on why these are set to valid experiments or valid exp results

File: tests/scripts/rm_hpo_api_tests.sh [hunk 2]
```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
...
+		failed=1
+		expected_result_="200"
+		expected_behaviour="RESPONSE_CODE = 200 OK"
+		((TOTAL_TESTS++))
+		((TESTS++))
+		error_message ${failed}
+	fi
+}
+
+# Other RM-HPO post result tests
```
Comment (chandrams) [inline tests/scripts/rm_hpo_api_tests.sh:446]: Elaborate on what is other tests

File: tests/scripts/rm_hpo_api_tests.sh [hunk 3]
```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
...
+		post_experiment_trial "${rm_hpo_post_exp_result_json[$exp_result]}"
+
+		actual_result="${http_code}"
+		expected_result_="^4[0-9][0-9]"
+		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+
+		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+	else
+		failed=1
+		expected_result_="200"
```
Comment (chandrams) [inline tests/scripts/rm_hpo_api_tests.sh:394]: Where are these used? Also there needs to be a message for the failure - "post experiment failed"

File: tests/scripts/rm_hpo_api_tests.sh [hunk 4]
```diff
@@ -328,8 +333,164 @@ function other_post_experiment_tests() {
...
```
Comment (chandrams) [inline tests/scripts/rm_hpo_api_tests.sh:438]: Where are expected_result and behaviour used? You need to log what failed - "Posting valid experiment failed"

File: tests/scripts/hpo_api_tests.sh [hunk 5]
```diff
@@ -133,17 +133,25 @@ function check_server_status() {
 	fi
 }
 
-# Do a post with JSON array having invalid fields
+# The test does the following:
+# In case of hpo_post_experiment test, Post valid and invalid experments to HPO /experiment_trials API and validate the reslut
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:137]: Correct the spelling of experiments

File: tests/scripts/hpo_api_tests.sh [hunk 6]
```diff
@@ -195,27 +217,27 @@ function invalid_post(){
...
+
 		echo "" | tee -a ${LOG_} ${LOG}
-	done
 
+	done
+	
 	echo "*********************************************************************************************************" | tee -a ${LOG_} ${LOG}
 }
 
 # Do a post on experiment_trials for the same id again with "operation: EXP_TRIAL_GENERATE_NEW" and check if experiments have started from the begining
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:237]: Change begining to beginning

File: tests/scripts/hpo_api_tests.sh [hunk 7]
```diff
@@ -226,73 +248,58 @@ function post_duplicate_experiments() {
...
 		
-		post_experiment_json "${rm_hpo_post_experiment_json[$exp]}"
+		post_experiment_json "${hpo_post_experiment_json[$exp]}"
 		
 		actual_result="${http_code}"
 		expected_result_="^4[0-9][0-9]"
 		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
 	
-		compare_result ${__test_name__} ${expected_result_} "${expected_behaviour}"
+		compare_result ${hpo_test_name} ${expected_result_} "${expected_behaviour}"
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:257]: Double quote all variables

File: tests/scripts/hpo_api_tests.sh [hunk 8]
```diff
@@ -226,73 +248,58 @@ function post_duplicate_experiments() {
...
 	else
 		failed=1
-		expected_result_="200"
 		expected_behaviour="RESPONSE_CODE = 200 OK"
-		((TOTAL_TESTS++))
-		((TESTS++))
-		error_message ${failed}
+		echo "Posting valid experiment failed"
+		display_result "${expected_behaviour}" ${hpo_test_name} ${failed}
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:262]: Double quote all variables

File: tests/scripts/hpo_api_tests.sh [hunk 9]
```diff
@@ -328,8 +335,161 @@ function other_post_experiment_tests() {
...
+	echo "" | tee -a ${LOG_} ${LOG}
+
+	echo "${post_result}" >> ${LOG_} ${LOG}
+
+	http_code=$(tail -n1 <<< "${post_result}")
+	response=$(echo -e "${post_result}" | tail -2 | head -1)
+	echo "Response is ${response}" >> ${LOG_} ${LOG}
+}
+
+# # Post duplicate experiment results to HPO /experiment_trials API and validate the result
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:360]: Remove a hash to maintain consistency

File: tests/scripts/hpo_api_tests.sh [hunk 10]
```diff
@@ -328,8 +335,161 @@ function other_post_experiment_tests() {
...
+		actual_result="${http_code}"
+		expected_result_="^4[0-9][0-9]"
+		expected_behaviour="RESPONSE_CODE = 4XX BAD REQUEST"
+
+		compare_result ${hpo_test_name} ${expected_result_} "${expected_behaviour}"
+	else
+		failed=1
+		expected_behaviour="RESPONSE_CODE = 200 OK"
+		echo "Posting valid experiment failed"
+		display_result "${expected_behaviour}" ${hpo_test_name} ${failed}
```
Comment (chandrams) [inline tests/scripts/hpo_api_tests.sh:398]: Double quote all the variables

Comment (chandrams) [reply tests/scripts/hpo_api_tests.sh:398]: Check the entire file and implement this please

**Review by chandrams — APPROVED** [review]
LGTM

**Comment by dinogun** [comment]
@chandrams Looks like there are some conflicts that need to be resolved.

**Pull request merged**